modules/http4s/test/src/ProtocolComplianceTest.scala

```diff
@@ -59,32 +58,41 @@ object ProtocolComplianceTest extends SimpleIOSuite {
 
   private val path = sys.env
     .get("MODEL_DUMP")
-    .map(Paths.get(_))
+    .map(fs2.io.file.Path(_))
     .getOrElse(sys.error("MODEL_DUMP env var not set"))
 
-  private val bytes: Array[Byte] = Files.readAllBytes(path)
-  private val doc = decodeDocument(bytes)
-  private val dynamicSchemaIndex =
-    loadDynamic(doc).getOrElse(sys.error("unable to load Dynamic path"))
+  private implicit val runtime: IORuntime = cats.effect.unsafe.IORuntime.global
+  private val dynamicSchemaIndexLoader: IO[DynamicSchemaIndex] = fs2.io.file
+    .Files[IO]
+    .readAll(path)
+    .compile
+    .toVector
+    .map(_.toArray)
+    .map(decodeDocument)
+    .map(loadDynamic(_).getOrElse(sys.error("unable to load Dynamic path")))
 
   val pizzaSpec = generateTests(ShapeId("alloy.test", "PizzaAdminService"))
 
-  pizzaSpec(dynamicSchemaIndex).foreach(tc =>
-    test(tc.name) {
-      tc.run
-        .map[Expectations] {
-          case Left(value) =>
-            Expectations.Helpers.failure(value)
-          case Right(_) =>
-            Expectations.Helpers.success
-        }
-        .attempt
-        .map {
-          case Right(expectations) => expectations
-          case Left(e)             => failure(e.getMessage)
+  dynamicSchemaIndexLoader
+    .map(
+      pizzaSpec(_).foreach(tc =>
+        test(tc.name) {
+          tc.run
+            .map[Expectations] {
+              case Left(value) =>
+                Expectations.Helpers.failure(value)
+              case Right(_) =>
+                Expectations.Helpers.success
+            }
+            .attempt
+            .map {
+              case Right(expectations) => expectations
+              case Left(e)             => failure(e.getMessage)
+            }
         }
-    }
-  )
+      )
+    )
+    .unsafeRunAndForget()
```

Comment:
Hah, interesting ... this is a sign we shouldn't use `SimpleIOSuite` but rather a higher-level suite-type, like : 

https://github.com/disneystreaming/weaver-test/blob/main/modules/core/src/weaver/suites.scala#L29 

Feel free to park this work until we can sync on this.

Reply:
@Baccata yeah , this is just a placeholder to get it working end to end
the file IO cross platform is the current blocker